File: storekit-offering-completing-and-restoring-in-app-purchases/Objective-C/macOS/PrimaryViewController.m
/*
See the LICENSE.txt file for this sample’s licensing information.

Abstract:
The base table view controller for sharing a table view between subclasses.
*/

#import "PrimaryViewController.h"

@implementation PrimaryViewController
#pragma mark - Reload UI

/// Reloads the UI with new products, invalid identifiers, and purchases.
-(void)reloadUIWithData:(NSMutableArray *)data {
    self.data = data;
    [self.tableView reloadData];
}

#pragma mark - NSTable​View​Data​Source

- (NSInteger)numberOfRowsInTableView:(NSTableView *)tableView {
    return self.data.count;
}

@end

File: storekit-offering-completing-and-restoring-in-app-purchases/Swift/macOS/PrimaryViewController.swift
/*
See the LICENSE.txt file for this sample’s licensing information.

Abstract:
The base table view controller for sharing a table view between subclasses.
*/

import Cocoa

class PrimaryViewController: NSViewController, NSTableViewDataSource, NSTableViewDelegate {
    // MARK: - Properties
    
    /// The data model that all PrimaryViewController subclasses use.
    var data = [Any]()
    
    /// The table view that all PrimaryViewController subclasses use.
    @IBOutlet weak var tableView: NSTableView!
    
    // MARK: - NSTable​View​Data​Source
    
    func numberOfRows(in tableView: NSTableView) -> Int {
        return data.count
    }
}

